en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+
+## Naming conventions
+
+### Casing
+
+- Use `camelCase` for variables, method names and parameters
+- Use `UpperCamelCase` for class names, interfaces, traits and enums
+- Use `snake_case` for private static configuration properties.
+- Use `SCREAMING_SNAKE_CASE` for constants including class contants, and environment variables
+- Note that there will be some inconsistencies with legacy code where static methods have been a combination of `MyClass::camelCase()` and `MyClass::snake_case()`
+
+### Accessors
+
+- For accessors omit the leading 'get' with getters and prefix with 'set' for setters. e.g. `function myValue():sting { return $this->value; }` and `function setMyValue(string $value): void { $this->value = $value; }`.
+- Note that there will be some inconsistencies with legacy code as get accessors have been a combination of "with get", "without get", and "without get with the first letter uppercased"
+
+### File suffixes and prefixes
+
+- This is for new classes and interfaces that are created. Legacy Silvesrtripe code is inconsistent whether or not it follows these conventions.
+- Suffix the classnames with the type of class that is subclassed. The suffixed is sometimes a shortened version of the parent class because it reads better while retaining easy comprehension. Here are some common examples with the parents classes in brackets
+  - `Exception` (Exception)
+  - `Controller` (Controller)
+  - `Extension` (Extension, DataExtension)
+  - `Admin` (ModelAdmin)
+  - `Handler` (RequestHandler)
+  - `Filter` (SearchFilter)
+  - `Form` (Form)
+  - `Field` (FormField)
+  - `Job` (QueuedJob)
+  - `Task` (BuildTask)
+  - `Service` (generic class type that has no parent class)
+Some parent classes have unorthodox naming, in which case suffix with something more standard
+  - `Page` (SiteTree)
+  - `Controller` (LeftAndMain, CMSMain)
+- Do not suffix classes if they are a subclass any of the following:
+  - `DataObject`
+- Suffix interfaces with `Interface`
+- Suffix traits with `Trait`
+- Prefix abstract classes with `Abstract`
+
+### Other naming conventions
+
+- Extensions should match the name of the class they are extending. For example, an extension class that's added to LeftAndMain should be called LeftAndMainExtension
+- When choosing names for variables and methods use the guiding principle "make it as obvious as possible"
+- Ensure that variable names and method naming is consistent. For example, if you are creating a new class and have a variable named `$itemValue`` which is passed to another method in the same class, the paramater name for that method should also be `$itemValue``.
+
+## Directory naming - src dir
+
+- Put similar types of classes in the same directory. For example, put Controllers in a Controllers directory
+- Put DataObjects in a directory called Models
+- Do not group different class types together in a directory relating to a single feature
```

Comment:
Yeah that example you gave makes sense

Basically what I want to avoid is the "TopPage" directory this is completely out of place in https://github.com/silverstripe/silverstripe-elemental/tree/5/src

Reply:
I don't think we should just put things in the root namespace because the module starts out small. You never know how big something will get, and it's nicer to be consistent even if it stays small. But that's not a discussion for this pr.

Reply:
@GuySartorelli That's fair.

Reply:
Updated